web/app/mu-plugins/filters.php

```diff
@@ -28,3 +28,58 @@
 add_filter( 'pantheon.multisite.config_filename', function ( $config_filename ) {
     return 'config/application.php';
 } );
+
+/**
+ * Correct core resource URLs for non-main sites in a subdirectory multisite network.
+ *
+ * @param string $url The URL to fix.
+ * @return string The fixed URL.
+ */
+function fix_core_resource_urls( $url ) {
+    $main_site_url = trailingslashit( network_site_url( '/' ) );
+    $current_site_path = trailingslashit( get_blog_details()->path );
+
+    // Parse the URL to get its components.
+    $parsed_url = parse_url( $url );
+
+    // If there is no path in the URL, return it as is.
+    if ( ! isset( $parsed_url['path'] ) || $parsed_url['path'] === '/' ) {
+        return $url;
+    }
+
+    // Extract the path from the parsed URL.
+    $path = $parsed_url['path'];
+
+    // Correct the path for core resources.
+    if ( strpos( $path, $current_site_path . 'wp-includes/' ) !== false ) {
+        $path = str_replace( $current_site_path . 'wp-includes/', 'wp-includes/', $path );
+    } elseif ( strpos( $path, $current_site_path . 'wp-admin/' ) !== false ) {
+        $path = str_replace( $current_site_path . 'wp-admin/', 'wp-admin/', $path );
+    } elseif ( strpos( $path, $current_site_path . 'wp-content/' ) !== false ) {
+        $path = str_replace( $current_site_path . 'wp-content/', 'wp-content/', $path );
+    } else {
+        return $url; // Return the original URL if no changes were needed.
+    }
```

Comment:
```suggestion
$core_paths = ['wp-includes/', 'wp-admin/', 'wp-content/'];
$path_modified = false;

foreach ($core_paths as $core_path) {
    if (strpos($path, $current_site_path . $core_path) !== false) {
        $path = str_replace($current_site_path . $core_path, $core_path, $path);
        $path_modified = true;
        break;
    }
}

// If the path was not modified, return the original URL.
if (!$path_modified) {
    return $url;
}
```

Not sure if we write test cases for these functions but this seems ripe for that.